### PR TITLE
fix #516: align trimmed chunks to 8 bytes

### LINKED ISSUE #516: Split chunks are not 8 bytes aligned
This can cause memory overflow, as seen in the following valgrind error:

```
==3872== 36532 errors in context 4 of 121:
==3872== Invalid write of size 8
==3872==    at 0x61792AC: appendBits (gorilla.c:218)
==3872==    by 0x61799D4: appendFloat (gorilla.c:348)
==3872==    by 0x6179BBD: Compressed_Append (gorilla.c:374)
==3872==    by 0x617622E: Compressed_AddSample (compressed_chunk.c:146)
==3872==    by 0x6175D93: ensureAddSample (compressed_chunk.c:46)
==3872==    by 0x6176192: Compressed_UpsertSample (compressed_chunk.c:133)
==3872==    by 0x618688F: SeriesUpsertSample (tsdb.c:306)
==3872==    by 0x617F95A: internalAdd (module.c:699)
==3872==    by 0x617FC73: add (module.c:756)
==3872==    by 0x617FE7C: TSDB_add (module.c:790)
==3872==    by 0x1D80D9: RedisModuleCommandDispatcher (module.c:644)
==3872==    by 0x147FE9: call (server.c:3368)
==3872==  Address 0x5c4ada0 is 304 bytes inside a block of size 308 alloc'd
==3872==    at 0x4C31B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==3872==    by 0x150524: zcalloc (zmalloc.c:122)
==3872==    by 0x1D79C1: RM_Calloc (module.c:404)
==3872==    by 0x6175BF5: Compressed_NewChunk (compressed_chunk.c:26)
==3872==    by 0x61760A7: Compressed_UpsertSample (compressed_chunk.c:108)
==3872==    by 0x618688F: SeriesUpsertSample (tsdb.c:306)
==3872==    by 0x617F95A: internalAdd (module.c:699)
==3872==    by 0x617FC73: add (module.c:756)
==3872==    by 0x617FE7C: TSDB_add (module.c:790)
==3872==    by 0x1D80D9: RedisModuleCommandDispatcher (module.c:644)
==3872==    by 0x147FE9: call (server.c:3368)
==3872==    by 0x19A778: execCommand (multi.c:208)
```

### PR DESCRIPTION
**Pull request by danni-m**

### CODE REVIEW

File: src/compressed_chunk.c [hunk 1]
```diff
@@ -62,6 +62,9 @@ static void trimChunk(CompressedChunk *chunk) {
 
     if (excess > 1) {
         size_t newSize = chunk->size - excess + 1;
+        // align to 8 bytes (u_int64_t) otherwise we will have an heap overflow in gorilla.c because each write happens
+        // in 8 bytes blocks.
+        newSize += newSize % 8;
```
Comment (ashtul) [inline src/compressed_chunk.c:67]:
```suggestion
        newSize += BIT - newSize % BIT;
```

**Review by ashtul — COMMENTED** [review]
nice catch
a small fix

File: src/compressed_chunk.c (hunk 1 above)
Comment (danni-m) [reply src/compressed_chunk.c:67]: its better to align to have more space not less.

File: src/compressed_chunk.c [hunk 2]
```diff
@@ -56,12 +56,15 @@ static void ensureAddSample(CompressedChunk *chunk, Sample *sample) {
...
-    int excess = (chunk->size * 8 - chunk->idx) / 8;
+    int excess = (chunk->size * BIT - chunk->idx) / BIT;
 
     assert(excess >= 0); // else we have written beyond allocated memory
 
     if (excess > 1) {
         size_t newSize = chunk->size - excess + 1;
+        // align to 8 bytes (u_int64_t) otherwise we will have an heap overflow in gorilla.c because
+        // each write happens in 8 bytes blocks.
+        newSize += sizeof(binary_t) - (newSize % sizeof(binary_t));
```
Comment (ashtul) [inline src/compressed_chunk.c:67]: we should either use `BIT` or sizeof(binary_t)
Maybe change `#define BIT 8` to `#define BIT sizeof(binary_t)`

Comment (danni-m) [reply src/compressed_chunk.c:67]: sizeof(binary_t) is in bytes not bits.

**Review by ashtul — APPROVED** [review]

**Pull request merged**